controller/src/controller/db/client.py

```diff
@@ -0,0 +1,688 @@
+# Copyright 2023 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC, abstractmethod
+from typing import List, Optional, Type, Union
+
+import genai_factory.schemas as api_models
+
+
+class Client(ABC):
+    """
+    Base class for all DB clients
+    In order to implement a new DB client, inherit from this class and implement all the methods.
+    """
+
+    @abstractmethod
+    def __init__(self):
+        pass
+
+    @abstractmethod
+    def get_local_session(self):
+        """
+        Get a local session from the local session maker.
+        This is the session that is inserted into the API endpoints.
+        :return: The session.
+        """
+        pass
+
+    @staticmethod
+    @abstractmethod
+    def _from_db_object(obj, obj_class):
```

Comment:
What about changing the name to
```suggestion
    def _to_schema_object(obj, schema: BaseOrWhateverWeHave):
```

Makes more sense as currently, I don't understand from what to what?

Reply:
this convert the db object into schema object